subsys/net/lib/download_client/src/download_client.c

```diff
@@ -409,14 +340,25 @@ static int client_connect(struct download_client *dl)
 		}
 	}
 
-	LOG_INF("Connecting to %s", dl->host);
+	if (IS_ENABLED(CONFIG_LOG)) {
+		char ip_addr_str[NET_IPV6_ADDR_LEN];
+		void *sin_addr;
+
+		if (dl->remote_addr.sa_family == AF_INET6) {
+			sin_addr = &((struct sockaddr_in6 *)&dl->remote_addr)->sin6_addr;
+		} else {
+			sin_addr = &((struct sockaddr_in *)&dl->remote_addr)->sin_addr;
+		}
+		inet_ntop(dl->remote_addr.sa_family, sin_addr, ip_addr_str, sizeof(ip_addr_str));
+		LOG_INF("Connecting to %s", ip_addr_str);
+	}
 	LOG_DBG("fd %d, addrlen %d, fam %s, port %d",
 		dl->fd, addrlen, str_family(dl->remote_addr.sa_family), port);
 
 	err = connect(dl->fd, &dl->remote_addr, addrlen);
 	if (err) {
+		LOG_ERR("Unable to connect, errno %d", errno);
 		err = -errno;
-		LOG_ERR("Unable to connect, errno %d", -err);
```

Comment:
Why switch the order? Is there guarantee that LOG() macros don't touch errno?

Reply:
This is the order used for other logs using errno.

Reply:
IMO the original order (read `errno` once before any other action) makes more sense if we want to align the codebase.